gridscale/resource_gridscale_server.go

```diff
@@ -655,6 +757,60 @@ func resourceGridscaleServerCreate(d *schema.ResourceData, meta interface{}) err
 		requestBody.HardwareProfile = gsclient.DefaultServerHardware
 	}
 
+	// Since only one hardware profile config can be set when can just use the index 0
+	if _, ok := d.GetOk("hardware_profile_config"); ok {
+		config := gsclient.ServerHardwareProfileConfig{
+			NestedVirtualization: d.Get("hardware_profile_config.0.nested_virtualization").(bool),
+			HyperVExtensions:     d.Get("hardware_profile_config.0.hyperv_extensions").(bool),
+			SerialInterface:      d.Get("hardware_profile_config.0.serial_interface").(bool),
+			ServerRenice:         d.Get("hardware_profile_config.0.server_renice").(bool),
+		}
+
+		machineType := d.Get("hardware_profile_config.0.machine_type").(string)
+		switch machineType {
+		case "i440fx":
```

Comment:
I've tried to mimic existing code like e.g. in the gridscale/resource_gridscale_storage.go file:

```go
//since only one template can be used, we can just look at index 0
	if _, ok := d.GetOk("template"); ok {
		template := gsclient.StorageTemplate{
			Password:     d.Get("template.0.password").(string),
			Hostname:     d.Get("template.0.hostname").(string),
			TemplateUUID: d.Get("template.0.template_uuid").(string),
```

I've changed the code to avoid magic those magic numbers.

Reply:
I think we could keep the other changes too.